### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by sumancrest0001**
Features of the project
1. Creating user, login, and logout
2. Listing all the attendees of a event.
3. Listing Past and Upcoming events for a user as well as in the index page.

### CODE REVIEW

**Comment by sumancrest0001** [comment]
Features of this project.
1. Creating users, login, and logout.
2. Listing attendees of an event.
3. Listing past and upcoming events on the index page.

File: app/controllers/user_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      sign_in @user
+      redirect_to @user
+    else
+      flash.now[:danger] = 'invalid inputs'
+      render 'new'
+    end
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
```
Comment (maelfosso) [inline app/controllers/user_controller.rb]: why not use `@current_user` ?
Screenshot from 2019-12-11 01-13-04

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+    <title><%= full_title(yield(:title)) %></title>
     <%= csrf_meta_tags %>
 
     <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
 
   <body>
+    <%= render 'layouts/navigation' %>
+    <%= render 'shared/flash_messages' %>
```
[the remark this answers is not in the document]
Comment (noomdalv) [reply app/views/layouts/application.html.erb]: Fixed, shared folder moved to the correct location

File: app/models/event.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  scope :past, -> { where('date < :current_time', current_time: DateTime.now) }
+  scope :upcoming, -> { where('date >= :current_time', current_time: DateTime.now) }
+
+  # def self.past
+  # where('date < :current_time', current_time: DateTime.now)
+  # end
+
+  # def self.upcoming
+  # where('date >= :current_time', current_time: DateTime.now)
+  # end
```
[the remark this answers is not in the document]
Comment (noomdalv) [reply app/models/event.rb]: Fixed, comments removed

File: app/controllers/user_controller.rb (hunk 1 above)
Comment (noomdalv) [reply app/controllers/user_controller.rb]: Fixed, @current_user is now used

**Pull request merged**